paywall/src/unlock.js/web3Proxy.ts

```diff
@@ -42,22 +110,104 @@ export default function web3Proxy(
     window.web3 &&
     window.web3.currentProvider &&
     (window.web3.currentProvider.sendAsync || window.web3.currentProvider.send)
+  let proxyAccount: null | string = null
+  let proxyNetwork: string | number
+  let accountIframeReady = false
+  let currentLocks: Locks
+  let canUseUserAccounts: boolean = !window.web3
+  let hasNativeWeb3Wallet = !!window.web3
+
+  const useUnlockAccount = () =>
+    !hasNativeWeb3Wallet && proxyAccount && canUseUserAccounts
```

Comment:
```suggestion
    proxyAccount && canUseUserAccounts
```

Reply:
this was the original code, but it turns out it does not work. There are some cases where we have a native wallet present and it should use that (for instance, when we have not yet enabled access to the page)

Reply:
Can you provide some more context? What are the cases where `canUseUserAccounts` and `!hasNativeWeb3Wallet` are different?